src/axom/primal/geometry/Hexahedron.hpp

```diff
@@ -0,0 +1,398 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef HEXAHEDRON_HPP_
+#define HEXAHEDRON_HPP_
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Tetrahedron.hpp"
+#include "axom/primal/geometry/Vector.hpp"
+#include "axom/primal/operators/squared_distance.hpp"
+
+#include "axom/slic/interface/slic.hpp"
+
+#include <ostream>  // for std::ostream
+
+namespace axom
+{
+namespace primal
+{
+/*!
+ * \class Hexahedron
+ *
+ * \brief Represents an hexahedral geometric shape defined by eight points.
+ *
+ * \accelerated
+ * \tparam T the coordinate type, e.g., double, float, etc.
+ * \tparam NDIMS the number of spatial dimensions
+ *
+ * There are eight vertices in the hexahedron, labelled P through W as the
+ * constructor's arguments.  They are accessible using the square-brackets
+ * operator, with P being index 0, Q index 1, through W as index 7.
+ *
+ * Here's a diagram showing
+ * a cube with the labeled vertices.
+ *
+ * \verbatim
+ *
+ * U +---------+ V              +y
+ *   |\        |\                    +z
+ *   |  \      |  \              ^  >
+ *   | T + --------+ W           | /
+ * Q +---|-----+ R |             |/
+ *   \   |     \   |             -----> +x
+ *    \  |      \  |
+ *     \ |       \ |
+ *   P  +----------+ S
+ *
+ * \endverbatim
```

Comment:
Thanks for the ascii art!

Suggestion: It might be clearer if the `z` axis has the same directions in the cube and triad.

Reply:
@rhornung67 Yes, it is left-handed.